lms/static/js/edxnotes/plugins/llm_summarize.js

```diff
@@ -68,17 +68,113 @@
                         background-color: white;
                         border: 1px solid gray;
                     }
+
+                    .summarize-loader-wrapper {
+                        display: none;
+                    }
+
+                    .summarize-loader-wrapper.show {
+                        display: block
+                    }
+                    // Defining content loader since fontawesome icons don't work
+                    // inside the annotatorjs modal.
+                    .loader {
+                        width: 5em !important;
+                        border-color: red !important;
+                    }
+                    @keyframes rotation {
+                    0% {
+                        transform: rotate(0deg);
+                    }
+                    100% {
+                        transform: rotate(360deg);
+                    }
                 `;
                 document.head.appendChild(style);
-                var annotator = this.annotator;
+                this.modifyDom(this.annotator);
+                const summarizeButton = document.getElementById('summarizeButton');
+                summarizeButton.addEventListener('click', function(ev) {
+                    document.querySelector('.annotator-outer.annotator-editor').setAttribute('is_summarizing', true);
+                });
+                this.annotator.subscribe('annotationEditorShown', this.handleSummarize);
+                this.annotator.subscribe('annotationEditorHidden', this.cleanupSummarize);
+            },
+            handleSummarize: function (editor, annotation) {
+                if (editor.element[0].getAttribute('is_summarizing') !== 'true') return;
```

Comment:
```suggestion
                if (editor.element[0].getAttribute('is_summarizing') === 'false') return;
```
Logic is the same, but I think it helps in readability. What do you think @alexjmpb ?

Reply:
The trouble with this is that when the attribute does not exist, the condition === 'false' is not met and the function is executed. This case happens when the summarize button has not been clicked and the user clicks the annotate pencil. This can be fixed by when the plugin initializes it sets the attribute to false. But I think it's better to leave it with the condition !== 'true'.